src/visualization/components/annotationController.tsx

```diff
@@ -0,0 +1,143 @@
+// React
```

Comment:
this file doesn't have any JSX in it, right? it should just be a ts file then.

for naming, how about just `annotations.ts` ? when I think of a controller or a manager, I think of a centralized object that delegates to sub objects or subroutines, whereas this is more of collection of similarly-concept functions

Reply:
i'll change it to annotationUtils.ts ; annotations is not description enough (i want to have a sense of what it does from the name)